File: perf_test.py
from time import time



all_primes = []

def func3(numMax):
    i = numMax
    isPrime = True
    for j in range(2, int(i ** 0.5) + 1):
        if i % j == 0:
            isPrime = False
            break

    return isPrime

def func3(numMax):
    i = numMax
    isPrime = True
    for j in range(2, int(i /2) + 1):
        if i % j == 0:
            isPrime = False
            break

    return isPrime


def func1(numMax):
    i = numMax
    isPrime = True
    for j in range(2, i):
        if i % j == 0:
            isPrime = False
            break

    return isPrime


def main():

    numMax = 10**7
    numMax = numMax + 3

    start = time()
    flag = func1(numMax)
    end = time()
    duration = end - start 
    print(f"Time taken: {duration:.8f} seconds  and {numMax} prime is {flag}")   

    start = time()
    flag = func3(numMax)
    end = time()
    duration = end - start 
    print(f"Time taken: {duration:.8f} seconds  and {numMax} prime is {flag}")   
        
    start = time()
    flag = func3(numMax)
    end = time()
    duration = end - start 
    print(f"Time taken: {duration:.8f} seconds  and {numMax} prime is {flag}")   


if __name__ == "__main__":
    main()

File: perf_test.java



// import java.util.ArrayList;
// import java.util.List;

public class perf_test {
    public static void main(String[] args) {
        int numMax = 100003;
        boolean isPrime = false;
        long startTime, endTime, duration;
        // List<Integer> allPrimes = new ArrayList<>();

        isPrime = false;
        startTime = System.nanoTime();
        isPrime = isPrime1(numMax);
        endTime = System.nanoTime();
        duration = endTime - startTime;
        System.out.println("1 - Time taken: " + duration + " nano seconds , and prime flag is " + isPrime);
        

        isPrime = false;
        startTime = System.nanoTime();
        isPrime = isPrime2(numMax);
        endTime = System.nanoTime();
        duration = endTime - startTime;
        System.out.println("2 - Time taken: " + duration + " nano seconds , and prime flag is " + isPrime);
        

        isPrime = false;
        startTime = System.nanoTime();
        isPrime = isPrime3(numMax);
        endTime = System.nanoTime();
        duration = endTime - startTime;
        System.out.println("3 - Time taken: " + duration + " nano seconds , and prime flag is " + isPrime);
        
    }




    public static boolean isPrime3(int num) {
        if (num < 2) {
            return false;
        }
        for (int j = 2; j <= Math.sqrt(num); j++) {
            if (num % j == 0) {
                return false;
            }
        }
        return true;
    }

    
    public static boolean isPrime2(int num) {
        if (num < 2) {
            return false;
        }
        for (int j = 2; j <= num/2 +1 ; j++) {
            if (num % j == 0) {
                return false;
            }
        }
        return true;
    }


    public static boolean isPrime1(int num) {
        if (num < 2) {
            return false;
        }
        for (int j = 2; j <= num; j++) {
            if (num % j == 0) {
                return false;
            }
        }
        return true;
    }    
}